jme3-desktop/src/main/java/com/jme3/system/NativeLibraries.java

```diff
@@ -0,0 +1,170 @@
+            .addNativeVariant(Platform.Windows_ARM64, "native/windows/arm64/bulletjme.dll", "bulletjme-arm64.dll")
+            .addNativeVariant(Platform.Linux32, "native/linux/x86/libbulletjme.so", "libbulletjme-x86.so")
+            .addNativeVariant(Platform.Linux64, "native/linux/x86_64/libbulletjme.so", "libbulletjme-x86_64.so")
+            .addNativeVariant(Platform.Linux_ARM32, "native/linux/arm32/libbulletjme.so", "libbulletjme-arm32.so")
+            .addNativeVariant(Platform.Linux_ARM64, "native/linux/arm64/libbulletjme.so", "libbulletjme-arm64.so")
+            .addNativeVariant(Platform.MacOSX32, "native/osx/x86/libbulletjme.dylib", "libbulletjme-x86.dylib")
+            .addNativeVariant(Platform.MacOSX64, "native/osx/x86_64/libbulletjme.dylib", "libbulletjme-x86_64.dylib")
+            .addNativeVariant(Platform.MacOSX_ARM64, "native/osx/arm64/libbulletjme.dylib", "libbulletjme-arm64.dylib")
+    ),
+
+    // JInput
+    // For OSX: Need to rename extension jnilib -> dylib when extracting
+    JInput(new LibraryInfo("jinput", libPath ->
+            // jinput handle loading by itself.
+            System.setProperty("net.java.games.input.librarypath",
+                    Paths.get(libPath).getParent().toAbsolutePath().toString()))
+            .addNativeVariant(Platform.Windows32, "jinput-raw.dll")
+            .addNativeVariant(Platform.Windows64, "jinput-raw_64.dll")
+            .addNativeVariant(Platform.Linux32, "libjinput-linux.so")
+            .addNativeVariant(Platform.Linux64, "libjinput-linux64.so")
+            .addNativeVariant(Platform.MacOSX32, "libjinput-osx.jnilib", "libjinput-osx.dylib")
+            .addNativeVariant(Platform.MacOSX64, "libjinput-osx.jnilib", "libjinput-osx.dylib")
+    ),
+
+    // JInput Auxiliary (only required on Windows)
+    JInputDX8(new LibraryInfo("jinput-dx8")
+            .addNativeVariant(Platform.Windows32, "jinput-dx8.dll", null)
+            .addNativeVariant(Platform.Windows64, "jinput-dx8_64.dll", null)
+            .addNativeVariant(Platform.Linux32, null)
+            .addNativeVariant(Platform.Linux64, null)
+            .addNativeVariant(Platform.MacOSX32, null)
+            .addNativeVariant(Platform.MacOSX64, null)
+    );
+
+    private final LibraryInfo library;
+
+
+    NativeLibraries(LibraryInfo library) {
+        this.library = library;
+    }
+
+    public static void registerDefaultLibraries() {
+        Lwjgl.registerLibrary();
+        Openal.registerLibrary();
+        BulletJme.registerLibrary();
+        JInput.registerLibrary();
+        JInputDX8.registerLibrary();
+    }
+
+    public LibraryInfo getLibrary() {
+        return library;
+    }
+
+    public String getName() {
+        return library.getName();
+    }
+
+    private void registerLibrary() {
+        library.getNativeVariants().forEach(NativeLibraryLoader::registerNativeLibrary);
+    }
+
+    public static class LibraryInfo {
```

Comment:
Could you please define javadocs for this class constructors and the public methods, please ?

Please, provide a full discriminative documentation including how the internal code process the [null] input at some parameters in brief.

Reply:
Done

Reply:
Looks good thanks.